src/main/scala/zio/sqs/producer/Producer.scala

```diff
@@ -103,11 +102,11 @@ object Producer {
     queueUrl: String,
     serializer: Serializer[T],
     settings: ProducerSettings = ProducerSettings()
-  ): ZManaged[R with Clock with Sqs, Throwable, Producer[T]] = {
+  ): ZIO[R with Clock with Sqs, Throwable, Producer[T]] = {
     val eventQueueSize = nextPower2(settings.batchSize * settings.parallelism)
     for {
-      eventQueue <- Queue.bounded[SqsRequestEntry[T]](eventQueueSize).toManagedWith(_.shutdown)
-      failQueue  <- Queue.bounded[SqsRequestEntry[T]](eventQueueSize).toManagedWith(_.shutdown)
+      eventQueue <- ZIO.scoped(ZIO.acquireRelease(Queue.bounded[SqsRequestEntry[T]](eventQueueSize))(_.shutdown))
```

Comment:
If you do that it will shut down the Queue right away. I think you have to include the scope in the return type of the function.

Reply:
Thanks!  Fixed.